### DISCUSSION TITLE
Query on Cognito auth code generated for React frontend

### DISCUSSION THREAD

**Discussion by himan085** [aws/aws-pdk#680] [Q&A] [answered]
When PDK generates code for the frontend React website, it checks if the function `getAuthenticatedUser` is defined or not: [code pointer](https://github.com/aws/aws-pdk/blob/f99b241183c6d0a631a17d077efdc84232196d65/packages/cloudscape-react-ts-website/samples/cloudscape-react-ts-website/src/layouts/App/index.tsx.mustache#L47)
This function is fetched from the context and used inside `useEffect` as shown below:
```
const { getAuthenticatedUser } = useCognitoAuthContext();

  useEffect(() => {
    if (getAuthenticatedUser) {
      const authUser = getAuthenticatedUser();
      // code doing stuff using authUser
    }
  }, [getAuthenticatedUser, setUsername, setEmail]);
```

I am not able to understand why the check `if (getAuthenticatedUser) {...}` is required. The `aws-northstar` library already provides a default state for the `CognitoAuthContext` which defines the `getAuthenticatedUser` method: [code pointer](https://github.com/aws/aws-northstar/blob/97528359596c4d988f48e3fce54a0a6bca8af9b3/packages/ui/src/components/CognitoAuth/context.ts#L62-L73) 

The same question applies to the [sign out code](https://github.com/aws/aws-pdk/blob/f99b241183c6d0a631a17d077efdc84232196d65/packages/cloudscape-react-ts-website/samples/cloudscape-react-ts-website/src/layouts/App/index.tsx.mustache#L106)

Please let me know if you have context on this, thanks in advance!

**Comment by agdimech** [comment, marked as answer]
Hi Himanshu!

This is a great question. The reason is that this code existed prior to [this PR](https://github.com/aws/aws-northstar/commit/214002276b285b0ba1f827c91bf3dff6c06e8575) in aws-northstar which changed the getAuthenticatorUser from optional to required.

You are 100% right though, these checks are no longer required given aws-northstar guarantees them to be present now - we just haven't changed the website code as it doesn't harm anything :)

Feel free to open a PR to remove this as that would be greatly appreciated!

**Reply by agdimech** [reply]
Thanks for contributing https://github.com/aws/aws-pdk/pull/683
